lte/gateway/python/magma/pipelined/app/inout.py

```diff
@@ -253,64 +255,75 @@ def _install_default_ingress_flows(self, dp):
                                                  match, actions=actions, priority=flows.DEFAULT_PRIORITY,
                                                  resubmit_table=next_table)
 
-    def _get_gw_mac_address(self, gw_ip: str) -> str:
+    def _get_gw_mac_address(self, ip: IPv4Address) -> str:
         try:
+            try:
+                gw_ip = ipaddress.ip_address(ip.address)
+                self.logger.info("GW ip address %s", gw_ip)
+            except ValueError:
+                self.logger.warning("Invalid GW Ip address: [%s]", ip)
+                return
+
             self.logger.debug("sending arp for IP: %s ovs egress: %s",
-                              gw_ip, self.config.non_nat_arp_egress_port)
-            eth_mac_src = get_if_hwaddr(self.config.non_nat_arp_egress_port)
+                              gw_ip, self.config.NON_NAT_ARP_EGRESS_PORT)
+            eth_mac_src = get_if_hwaddr(self.config.NON_NAT_ARP_EGRESS_PORT)
 
             pkt = Ether(dst=ETHER_BROADCAST, src=eth_mac_src)
             pkt /= ARP(op="who-has", pdst=gw_ip, hwsrc=eth_mac_src, psrc="0.0.0.0")
-            self.logger.debug("pkt: %s", pkt.summary())
+            self.logger.debug("ARP Req pkt %s", pkt.show(dump=True))
 
             res = srp1(pkt,
                        type=ETH_P_ARP,
-                       iface=self.config.non_nat_arp_egress_port,
+                       iface=self.config.NON_NAT_ARP_EGRESS_PORT,
                        timeout=1,
                        verbose=0,
                        nofilter=1,
                        promisc=0)
 
             if res is not None:
-                self.logger.debug("resp: %s ", res.summary())
+                self.logger.debug("ARP Res pkt %s", res.show(dump=True))
+
                 mac = res[ARP].hwsrc
                 return mac
             else:
-                self.logger.debug("Got Null response")
+                self.logger.info("Got Null response")
 
         except Scapy_Exception as ex:
             self.logger.warning("Error in probing Mac address: err %s", ex)
             return None
 
     def _monitor_and_update(self, datapath):
-        current_feq = self.config.non_mat_gw_probe_frequency
-        if self._dhcp_gw_info.getMac() is not None:
-            latest_mac_addr = self._dhcp_gw_info.getMac()
-            self._install_default_egress_flows(datapath, latest_mac_addr)
-            flows.set_barrier(datapath)
+        while True:
+            gw_info = get_mobilityd_gw_info()
+            if gw_info is None:
+                self.logger.warning("No gw_info")
+            else:
+                break
+            threading.Event().wait(timeout=self.config.non_mat_gw_probe_frequency)
+
+        if self._current_upstream_mac is not None and \
+                len(self._current_upstream_mac) != 17:
```

Comment:
If its not none but mailformed(len(self._current_upstream_mac) != 17) we should log something?